prov/sm2/src/sm2_ep.c

```diff
@@ -228,6 +261,7 @@ static void cleanup_shm_resources(struct sm2_ep *ep)
 		} else {
 			/* TODO Tell other side that we haven't processed their
 			 * message, just returned xfer_entry */
+			xfer_entry->hdr.proto_flags &= ~SM2_GENERATE_COMPLETION;
```

Comment:
I think this should be in a different commit, and that we should add an error flag to generate error completion.

Reply:
done